### DISCUSSION TITLE
What happened to `vim.o.term`?

### DISCUSSION THREAD

**Discussion by itaranto** [neovim/neovim#28982] [Q&A]
I used to be able to detect if running in a TTY with `vim.o.term` in 0.9.x but it seems to have been removed in 0.10.0.

Is this documented anywhere?

**Comment by wookayin** [comment]
`vim.o` is supposed to have the keys as 'options' would do. ~~'term' is not a valid option, see `:help 'term'` or try `:set term?` Therefore, `vim.o.term` reporting the same as `$TERM` was more like a bug in previous versions: in NVIM 0.9.x `nvim_get_option('term')` used to return something, but in NVIM 0.10 it raises an error. Same for `ttytype`.~~ EDIT: `set term?` doesn't work, but `set term`, `set noterm` works (though it's strange because it's a read-only option), so it's an option that should be recognized. Somehow such options were not included in the options-list.

Telling from git bisect, these options were removed in 6c87d3e0f (#25716), but I'm not sure if it was intended because the change was neither discussed nor mentioned explicitly. Another relevant PR: #26512. /cc @faimu @bfredl

**Comment by bfredl** [comment]
it is supposed to be consistent with with vim script. as `:echo &term` does work, we would expect `:lua print(vim.o.term)` to work as well.

**Reply by itaranto** [reply]
So, are you saying this is a bug?
